File: app/src/main/java/blog/david/ojeda/kotlin_android_extensions/MainActivity.java
package blog.david.ojeda.kotlin_android_extensions;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.Switch;
import android.widget.TextView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText editText = (EditText) findViewById(R.id.editText);
        editText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                // Do something...
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                // Do something...
            }

            @Override
            public void afterTextChanged(Editable s) {
                // Do something...
            }
        });

        Button button1 = (Button) findViewById(R.id.button);
        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Do something...
            }
        });

        Switch switch1 = (Switch) findViewById(R.id.switch1);
        switch1.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {

            @Override
            public void onCheckedChanged(CompoundButton buttonView,
                                         boolean isChecked) {
                // Do something...
            }
        });

        Switch switch2 = (Switch) findViewById(R.id.switch2);
        switch2.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {

            @Override
            public void onCheckedChanged(CompoundButton buttonView,
                                         boolean isChecked) {
                // Do something...
            }
        });

        Switch switch3 = (Switch) findViewById(R.id.switch3);
        switch3.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {

            @Override
            public void onCheckedChanged(CompoundButton buttonView,
                                         boolean isChecked) {
                // Do something...
            }
        });

        TextView textView = (TextView) findViewById(R.id.textView);
        textView.setText("Woh, a ListView!");


        ArrayList<String> items = new ArrayList<>();
        items.add("first");
        items.add("second");
        items.add("third");
        items.add("fourth");
        items.add("fifth");

        ListAdapter adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, items);
        ListView listView = (ListView) findViewById(R.id.listView);
        listView.setAdapter(adapter);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener()
        {
            @Override
            public void onItemClick(AdapterView<?> adapter, View v, int position,
                                    long arg3)
            {
               // Do something...
            }
        });

        Button button2 = (Button) findViewById(R.id.button2);
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Do something...
            }
        });


    }
}

File: app/src/main/kotlin/blog/david/ojeda/kotlin_android_extensions/MainActivity.kt
package blog.david.ojeda.kotlin_android_extensions

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.Editable
import android.text.TextWatcher
import android.widget.AdapterView
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editText!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
                // Do something...
            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                // Do something...
            }

            override fun afterTextChanged(s: Editable) {
                // Do something...
            }
        })

        button!!.setOnClickListener {
            // Do something...
        }

        switch1!!.setOnCheckedChangeListener { buttonView, isChecked ->
            // Do something...
        }

        switch2!!.setOnCheckedChangeListener { buttonView, isChecked ->
            // Do something...
        }

        switch3!!.setOnCheckedChangeListener { buttonView, isChecked ->
            // Do something...
        }

        textView!!.text = "Woh, a ListView!"

        val items = ArrayList<String>()
        items.add("first")
        items.add("second")
        items.add("third")
        items.add("fourth")
        items.add("fifth")

        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, items)
        listView!!.adapter = adapter
        listView.onItemClickListener = AdapterView.OnItemClickListener {
            adapter, v, position, arg3 -> // Do something...
        }

        button2!!.setOnClickListener {
            // Do something...
        }
    }
}
